src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -3,37 +3,55 @@
 #include <cos_debug.h>
 #include <cos_types.h>
 #include <llprint.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+#include <sl.h>
+#include <llboot.h>
+#include <res_spec.h>
 
 #define UNDEF_SYMBS 64
 
 /* Assembly function for sinv from new component */
-extern void *__inv_test_entry(int a, int b, int c);
+extern void *llboot_entry_inv(int a, int b, int c);
+extern int num_cobj;
+extern int resmgr_spdid;
+extern int root_spdid;
 
 struct cobj_header *hs[MAX_NUM_SPDS + 1];
 
 /* The booter uses this to keep track of each comp */
 struct comp_cap_info {
 	struct cos_compinfo *compinfo;
+	struct cos_defcompinfo *defci;
+
+	struct cos_defcompinfo def_cinfo;
 	struct usr_inv_cap   ST_user_caps[UNDEF_SYMBS];
 	vaddr_t              vaddr_user_caps; // vaddr of user caps table in comp
 	vaddr_t              addr_start;
 	vaddr_t              vaddr_mapped_in_booter;
 	vaddr_t              upcall_entry;
+	int                  is_sched;
+	int                  sched_no;
+	int                  parent_no;
+	spdid_t              parent_spdid;
+	u64_t                childid_bitf;
+	u64_t                childid_sched_bitf;
+	struct sl_thd       *initaep;
 } new_comp_cap_info[MAX_NUM_SPDS + 1];
 
-struct cos_compinfo boot_info;
-struct cos_compinfo new_compinfo[MAX_NUM_SPDS + 1];
+struct cos_compinfo *boot_info;
```

Comment:
Didn't @RobertGiff literally just get rid of this in his PR?

Reply:
Got rid of it(global boot_info pointer) entirely!